runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/internal/watcher/WatchedItem.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2021-2023 Aklivity Inc.
+ *
+ * Aklivity licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.aklivity.zilla.runtime.engine.internal.watcher;
+
+import static java.nio.file.StandardWatchEventKinds.ENTRY_CREATE;
+import static java.nio.file.StandardWatchEventKinds.ENTRY_DELETE;
+import static java.nio.file.StandardWatchEventKinds.ENTRY_MODIFY;
+import static org.agrona.LangUtil.rethrowUnchecked;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.WatchKey;
+import java.nio.file.WatchService;
+import java.util.Arrays;
+import java.util.Deque;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.Set;
+
+public class WatchedItem
+{
+    private final WatchService watchService;
+    private final Set<WatchKey> watchKeys;
+    private final Path watchedPath;
+    private byte[] hash;
+
+    public WatchedItem(
+        Path watchedPath,
+        WatchService watchService)
+    {
+        this.watchService = watchService;
+        this.watchKeys = new HashSet<>();
+        this.watchedPath = watchedPath;
+    }
+
+    public Set<WatchKey> keys()
+    {
+        return watchKeys;
+    }
+
+    public void register()
+    {
+        try
+        {
+            String scheme = watchedPath.getFileSystem().provider().getScheme();
+            if ("file".equals(scheme))
+            {
+                registerFilePath();
+            }
+            else if ("http".equals(scheme))
+            {
+                watchKeys.add(watchedPath.register(watchService));
+            }
```

Comment:
There is different behaviour here.

In case of the file filesystem we want to watch files, but actually we can only watch directories. But our file can be a symlink or the directory can be symlink or its parent etc... So Barbabas created a very sophisticated logic to cover all those cases, which means in a generic case we have to watch more than one object (directory) so we can make sure we are watching the changes of our file.

In case of http filesystem we just watch a URL and that's it. So it's a 1:1 mapping hence just one line of code here.